src/PluginClient.php

```diff
@@ -57,11 +57,43 @@ public function __construct($client, array $plugins = [], array $options = [])
             throw new \RuntimeException('Client must be an instance of Http\\Client\\HttpClient or Http\\Client\\HttpAsyncClient');
         }
 
-        $this->plugins = $plugins;
+        $this->plugins = $this->setPlugins($plugins);
         $this->options = $this->configure($options);
     }
 
     /**
+     * Append a plugin to the end of the queue.
+     *
+     * @param Plugin $plugin
+     */
+    public function addPlugin(Plugin $plugin)
+    {
+        $this->plugins[] = $plugin;
+    }
+
+    /**
+     * Get all active plugins.
+     *
+     * @return Plugin[]
+     */
+    public function getPlugins()
+    {
+        return $this->plugins;
+    }
+
+    /**
+     * Assign plugins to the client.
+     *
+     * @param Plugin[] $plugins
+     */
+    public function setPlugins(array $plugins = [])
```

Comment:
this should be appendPlugins, setPlugins makes me expect that this would replace the existing plugins (which would be a bad idea)